cluster-autoscaler/estimator/binpacking_estimator.go

```diff
@@ -105,47 +110,66 @@ func (e *BinpackingNodeEstimator) Estimate(
 			newNodesWithPods[nodeName] = true
 		}
 
-		if !found {
-			// If the last node we've added is empty and the pod couldn't schedule on it, it wouldn't be able to schedule
-			// on a new node either. There is no point adding more nodes to snapshot in such case, especially because of
-			// performance cost each extra node adds to future FitsAnyNodeMatching calls.
-			if lastNodeName != "" && !newNodesWithPods[lastNodeName] {
-				continue
+		for ; index < len(podsEquivalenceGroup.Pods); index++ {
+			pod := podsEquivalenceGroup.Pods[index]
+			found := false
+
+			// Check schedulability on only newly created node
+			nodeName, err := e.predicateChecker.FitsAnyNodeMatching(e.clusterSnapshot, pod, func(nodeInfo *schedulerframework.NodeInfo) bool {
```

Comment:
Since we only want to check one node here, perhaps it would suffice to call `CheckPredicates` instead? That'll avoid listing all nodes from cluster snapshot just to pick one of them later on. Also, no point in running this check when `lastNodeName` is empty I think?

Reply:
Make sense, rewrote.